addons/mail/models/res_partner.py

```diff
@@ -83,6 +83,8 @@ def find_or_create(self, email, assert_valid_email=False):
             raise ValueError(_('An email is required for find_or_create to work'))
 
         parsed_name, parsed_email = self._parse_partner_name(email)
+        if not parsed_email and assert_valid_email:
+            raise ValueError(_('A valid email is required to create a new customer.'))
```

Comment:
Actually, couldn't we say "%(email)s is not recognized as a valid email. This is required to create a new customer." ? That way we add the email itself in the message, as sometimes it is not easy to know where it does come from.